### ISSUE TITLE
Collectionchanges processing differences between PLocal and Memory Storage

### ISSUE THREAD

**Issue by GoorMoon** [orientechnologies/orientdb#3417]
Hello, 
I am tackle some issue in orientdb version 1.7.9
(Maybe all 1.7.* version have same issue)
all commands in binary protocol that return collection-changes behave different.
See console debug output for REQUEST_TX_COMMIT command:
**PLOCAL**
```
2015-01-17 13:38:56:939 INFO /10.0.2.2:53349 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:38:56:945 INFO /10.0.2.2:53349 - Read byte: 60 [OChannelBinaryServer]
2015-01-17 13:38:56:945 INFO /10.0.2.2:53349 - Reading int (4 bytes)... [OChannelBinaryServer]
2015-01-17 13:38:56:946 INFO /10.0.2.2:53349 - Read int: 4 [OChannelBinaryServer]
2015-01-17 13:38:56:949 INFO /10.0.2.2:53349 - Reading int (4 bytes)... [OChannelBinaryServer]
2015-01-17 13:38:56:949 INFO /10.0.2.2:53349 - Read int: 1 [OChannelBinaryServer]
2015-01-17 13:38:56:949 INFO /10.0.2.2:53349 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:38:56:949 INFO /10.0.2.2:53349 - Read byte: 0 [OChannelBinaryServer]
2015-01-17 13:38:56:949 INFO /10.0.2.2:53349 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:38:56:949 INFO /10.0.2.2:53349 - Read byte: 1 [OChannelBinaryServer]
2015-01-17 13:38:56:949 INFO /10.0.2.2:53349 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:38:56:949 INFO /10.0.2.2:53349 - Read byte: 3 [OChannelBinaryServer]
2015-01-17 13:38:56:950 INFO /10.0.2.2:53349 - Reading short (2 bytes)... [OChannelBinaryServer]
2015-01-17 13:38:56:950 INFO /10.0.2.2:53349 - Read short: 11 [OChannelBinaryServer]
2015-01-17 13:38:56:950 INFO /10.0.2.2:53349 - Reading cluster position (8 bytes).... [OChannelBinaryServer]
2015-01-17 13:38:56:950 INFO /10.0.2.2:53349 - Read cluster position: -2 [OChannelBinaryServer]
2015-01-17 13:38:56:950 INFO /10.0.2.2:53349 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:38:56:950 INFO /10.0.2.2:53349 - Read byte: 100 [OChannelBinaryServer]
2015-01-17 13:38:56:951 INFO /10.0.2.2:53349 - Reading chunk of bytes. Reading chunk length as int (4 bytes)... [OChannelBinaryServer]
2015-01-17 13:38:56:951 INFO /10.0.2.2:53349 - Read chunk lenght: 48 [OChannelBinaryServer]
2015-01-17 13:38:56:959 INFO /10.0.2.2:53349 - Reading 48 bytes... [OChannelBinaryServer]
2015-01-17 13:38:56:960 INFO /10.0.2.2:53349 - Read 48 bytes: TestVertex@Name:"First",Bar:1,in_TestEdge:#12:-4 [OChannelBinaryServer]
2015-01-17 13:38:56:960 INFO /10.0.2.2:53349 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:38:56:960 INFO /10.0.2.2:53349 - Read byte: 1 [OChannelBinaryServer]
2015-01-17 13:38:56:960 INFO /10.0.2.2:53349 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:38:56:960 INFO /10.0.2.2:53349 - Read byte: 3 [OChannelBinaryServer]
2015-01-17 13:38:56:960 INFO /10.0.2.2:53349 - Reading short (2 bytes)... [OChannelBinaryServer]
2015-01-17 13:38:56:961 INFO /10.0.2.2:53349 - Read short: 11 [OChannelBinaryServer]
2015-01-17 13:38:56:961 INFO /10.0.2.2:53349 - Reading cluster position (8 bytes).... [OChannelBinaryServer]
2015-01-17 13:38:56:961 INFO /10.0.2.2:53349 - Read cluster position: -3 [OChannelBinaryServer]
2015-01-17 13:38:56:961 INFO /10.0.2.2:53349 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:38:56:961 INFO /10.0.2.2:53349 - Read byte: 100 [OChannelBinaryServer]
2015-01-17 13:38:56:961 INFO /10.0.2.2:53349 - Reading chunk of bytes. Reading chunk length as int (4 bytes)... [OChannelBinaryServer]
2015-01-17 13:38:56:961 INFO /10.0.2.2:53349 - Read chunk lenght: 50 [OChannelBinaryServer]
2015-01-17 13:38:56:962 INFO /10.0.2.2:53349 - Reading 50 bytes... [OChannelBinaryServer]
2015-01-17 13:38:56:962 INFO /10.0.2.2:53349 - Read 50 bytes: TestVertex@Name:"Second",Bar:2,out_TestEdge:#12:-4 [OChannelBinaryServer]
2015-01-17 13:38:56:962 INFO /10.0.2.2:53349 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:38:56:962 INFO /10.0.2.2:53349 - Read byte: 1 [OChannelBinaryServer]
2015-01-17 13:38:56:962 INFO /10.0.2.2:53349 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:38:56:962 INFO /10.0.2.2:53349 - Read byte: 3 [OChannelBinaryServer]
2015-01-17 13:38:56:962 INFO /10.0.2.2:53349 - Reading short (2 bytes)... [OChannelBinaryServer]
2015-01-17 13:38:56:962 INFO /10.0.2.2:53349 - Read short: 12 [OChannelBinaryServer]
2015-01-17 13:38:56:971 INFO /10.0.2.2:53349 - Reading cluster position (8 bytes).... [OChannelBinaryServer]
2015-01-17 13:38:56:971 INFO /10.0.2.2:53349 - Read cluster position: -4 [OChannelBinaryServer]
2015-01-17 13:38:56:971 INFO /10.0.2.2:53349 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:38:56:971 INFO /10.0.2.2:53349 - Read byte: 100 [OChannelBinaryServer]
2015-01-17 13:38:56:972 INFO /10.0.2.2:53349 - Reading chunk of bytes. Reading chunk length as int (4 bytes)... [OChannelBinaryServer]
2015-01-17 13:38:56:972 INFO /10.0.2.2:53349 - Read chunk lenght: 41 [OChannelBinaryServer]
2015-01-17 13:38:56:972 INFO /10.0.2.2:53349 - Reading 41 bytes... [OChannelBinaryServer]
2015-01-17 13:38:56:972 INFO /10.0.2.2:53349 - Read 41 bytes: TestEdge@Weight:1.3f,in:#11:-2,out:#11:-3 [OChannelBinaryServer]
2015-01-17 13:38:56:972 INFO /10.0.2.2:53349 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:38:56:972 INFO /10.0.2.2:53349 - Read byte: 0 [OChannelBinaryServer]
2015-01-17 13:38:56:972 INFO /10.0.2.2:53349 - Reading chunk of bytes. Reading chunk length as int (4 bytes)... [OChannelBinaryServer]
2015-01-17 13:38:56:972 INFO /10.0.2.2:53349 - Read chunk lenght: 0 [OChannelBinaryServer]
2015-01-17 13:38:56:973 INFO /10.0.2.2:53349 - Reading 0 bytes... [OChannelBinaryServer]
2015-01-17 13:38:56:973 INFO /10.0.2.2:53349 - Read 0 bytes:  [OChannelBinaryServer]
2015-01-17 13:38:56:999 INFO /10.0.2.2:53349 - Writing byte (1 byte): 0 [OChannelBinaryServer]
2015-01-17 13:38:57:000 INFO /10.0.2.2:53349 - Writing int (4 bytes): 4 [OChannelBinaryServer]
2015-01-17 13:38:57:000 INFO /10.0.2.2:53349 - Writing int (4 bytes): 3 [OChannelBinaryServer]
2015-01-17 13:38:57:000 INFO /10.0.2.2:53349 - Writing short (2 bytes): 11 [OChannelBinaryServer]
2015-01-17 13:38:57:000 INFO /10.0.2.2:53349 - Writing cluster position (8 bytes) : -3.... [OChannelBinaryServer]
2015-01-17 13:38:57:000 INFO /10.0.2.2:53349 - Writing short (2 bytes): 11 [OChannelBinaryServer]
2015-01-17 13:38:57:001 INFO /10.0.2.2:53349 - Writing cluster position (8 bytes) : 1.... [OChannelBinaryServer]
2015-01-17 13:38:57:001 INFO /10.0.2.2:53349 - Writing short (2 bytes): 11 [OChannelBinaryServer]
2015-01-17 13:38:57:001 INFO /10.0.2.2:53349 - Writing cluster position (8 bytes) : -2.... [OChannelBinaryServer]
2015-01-17 13:38:57:001 INFO /10.0.2.2:53349 - Writing short (2 bytes): 11 [OChannelBinaryServer]
2015-01-17 13:38:57:001 INFO /10.0.2.2:53349 - Writing cluster position (8 bytes) : 0.... [OChannelBinaryServer]
2015-01-17 13:38:57:001 INFO /10.0.2.2:53349 - Writing short (2 bytes): 12 [OChannelBinaryServer]
2015-01-17 13:38:57:001 INFO /10.0.2.2:53349 - Writing cluster position (8 bytes) : -4.... [OChannelBinaryServer]
2015-01-17 13:38:57:009 INFO /10.0.2.2:53349 - Writing short (2 bytes): 12 [OChannelBinaryServer]
2015-01-17 13:38:57:009 INFO /10.0.2.2:53349 - Writing cluster position (8 bytes) : 0.... [OChannelBinaryServer]
2015-01-17 13:38:57:009 INFO /10.0.2.2:53349 - Writing int (4 bytes): 3 [OChannelBinaryServer]
2015-01-17 13:38:57:010 INFO /10.0.2.2:53349 - Writing short (2 bytes): 12 [OChannelBinaryServer]
2015-01-17 13:38:57:010 INFO /10.0.2.2:53349 - Writing cluster position (8 bytes) : 0.... [OChannelBinaryServer]
2015-01-17 13:38:57:011 INFO /10.0.2.2:53349 - Writing int (4 bytes): 2 [OChannelBinaryServer]
2015-01-17 13:38:57:011 INFO /10.0.2.2:53349 - Writing short (2 bytes): 11 [OChannelBinaryServer]
2015-01-17 13:38:57:011 INFO /10.0.2.2:53349 - Writing cluster position (8 bytes) : 1.... [OChannelBinaryServer]
2015-01-17 13:38:57:011 INFO /10.0.2.2:53349 - Writing int (4 bytes): 1 [OChannelBinaryServer]
2015-01-17 13:38:57:011 INFO /10.0.2.2:53349 - Writing short (2 bytes): 11 [OChannelBinaryServer]
2015-01-17 13:38:57:011 INFO /10.0.2.2:53349 - Writing cluster position (8 bytes) : 0.... [OChannelBinaryServer]
2015-01-17 13:38:57:011 INFO /10.0.2.2:53349 - Writing int (4 bytes): 2 [OChannelBinaryServer]
2015-01-17 13:38:57:012 INFO /10.0.2.2:53349 - Writing int (4 bytes): 0 [OChannelBinaryServer]
2015-01-17 13:38:57:012 INFO /10.0.2.2:53349 - Flush [OChannelBinaryServer]
2015-01-17 13:38:57:012 INFO /10.0.2.2:53349 - Reading byte (1 byte)... [OChannelBinaryServer]
```

**MEMORY**
```
2015-01-17 13:41:04:884 INFO /10.0.2.2:53356 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:41:04:902 INFO /10.0.2.2:53356 - Read byte: 60 [OChannelBinaryServer]
2015-01-17 13:41:04:902 INFO /10.0.2.2:53356 - Reading int (4 bytes)... [OChannelBinaryServer]
2015-01-17 13:41:04:902 INFO /10.0.2.2:53356 - Read int: 4 [OChannelBinaryServer]
2015-01-17 13:41:04:912 INFO /10.0.2.2:53356 - Reading int (4 bytes)... [OChannelBinaryServer]
2015-01-17 13:41:04:912 INFO /10.0.2.2:53356 - Read int: 1 [OChannelBinaryServer]
2015-01-17 13:41:04:917 INFO /10.0.2.2:53356 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:41:04:917 INFO /10.0.2.2:53356 - Read byte: 0 [OChannelBinaryServer]
2015-01-17 13:41:04:917 INFO /10.0.2.2:53356 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:41:04:917 INFO /10.0.2.2:53356 - Read byte: 1 [OChannelBinaryServer]
2015-01-17 13:41:04:917 INFO /10.0.2.2:53356 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:41:04:918 INFO /10.0.2.2:53356 - Read byte: 3 [OChannelBinaryServer]
2015-01-17 13:41:04:918 INFO /10.0.2.2:53356 - Reading short (2 bytes)... [OChannelBinaryServer]
2015-01-17 13:41:04:918 INFO /10.0.2.2:53356 - Read short: 11 [OChannelBinaryServer]
2015-01-17 13:41:04:918 INFO /10.0.2.2:53356 - Reading cluster position (8 bytes).... [OChannelBinaryServer]
2015-01-17 13:41:04:918 INFO /10.0.2.2:53356 - Read cluster position: -2 [OChannelBinaryServer]
2015-01-17 13:41:04:918 INFO /10.0.2.2:53356 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:41:04:918 INFO /10.0.2.2:53356 - Read byte: 100 [OChannelBinaryServer]
2015-01-17 13:41:04:919 INFO /10.0.2.2:53356 - Reading chunk of bytes. Reading chunk length as int (4 bytes)... [OChannelBinaryServer]
2015-01-17 13:41:04:919 INFO /10.0.2.2:53356 - Read chunk lenght: 48 [OChannelBinaryServer]
2015-01-17 13:41:04:919 INFO /10.0.2.2:53356 - Reading 48 bytes... [OChannelBinaryServer]
2015-01-17 13:41:04:919 INFO /10.0.2.2:53356 - Read 48 bytes: TestVertex@Name:"First",Bar:1,in_TestEdge:#12:-4 [OChannelBinaryServer]
2015-01-17 13:41:04:919 INFO /10.0.2.2:53356 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:41:04:919 INFO /10.0.2.2:53356 - Read byte: 1 [OChannelBinaryServer]
2015-01-17 13:41:04:919 INFO /10.0.2.2:53356 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:41:04:928 INFO /10.0.2.2:53356 - Read byte: 3 [OChannelBinaryServer]
2015-01-17 13:41:04:928 INFO /10.0.2.2:53356 - Reading short (2 bytes)... [OChannelBinaryServer]
2015-01-17 13:41:04:928 INFO /10.0.2.2:53356 - Read short: 11 [OChannelBinaryServer]
2015-01-17 13:41:04:928 INFO /10.0.2.2:53356 - Reading cluster position (8 bytes).... [OChannelBinaryServer]
2015-01-17 13:41:04:928 INFO /10.0.2.2:53356 - Read cluster position: -3 [OChannelBinaryServer]
2015-01-17 13:41:04:928 INFO /10.0.2.2:53356 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:41:04:929 INFO /10.0.2.2:53356 - Read byte: 100 [OChannelBinaryServer]
2015-01-17 13:41:04:929 INFO /10.0.2.2:53356 - Reading chunk of bytes. Reading chunk length as int (4 bytes)... [OChannelBinaryServer]
2015-01-17 13:41:04:929 INFO /10.0.2.2:53356 - Read chunk lenght: 50 [OChannelBinaryServer]
2015-01-17 13:41:04:929 INFO /10.0.2.2:53356 - Reading 50 bytes... [OChannelBinaryServer]
2015-01-17 13:41:04:929 INFO /10.0.2.2:53356 - Read 50 bytes: TestVertex@Name:"Second",Bar:2,out_TestEdge:#12:-4 [OChannelBinaryServer]
2015-01-17 13:41:04:929 INFO /10.0.2.2:53356 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:41:04:929 INFO /10.0.2.2:53356 - Read byte: 1 [OChannelBinaryServer]
2015-01-17 13:41:04:930 INFO /10.0.2.2:53356 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:41:04:930 INFO /10.0.2.2:53356 - Read byte: 3 [OChannelBinaryServer]
2015-01-17 13:41:04:930 INFO /10.0.2.2:53356 - Reading short (2 bytes)... [OChannelBinaryServer]
2015-01-17 13:41:04:930 INFO /10.0.2.2:53356 - Read short: 12 [OChannelBinaryServer]
2015-01-17 13:41:04:930 INFO /10.0.2.2:53356 - Reading cluster position (8 bytes).... [OChannelBinaryServer]
2015-01-17 13:41:04:930 INFO /10.0.2.2:53356 - Read cluster position: -4 [OChannelBinaryServer]
2015-01-17 13:41:04:930 INFO /10.0.2.2:53356 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:41:04:930 INFO /10.0.2.2:53356 - Read byte: 100 [OChannelBinaryServer]
2015-01-17 13:41:04:935 INFO /10.0.2.2:53356 - Reading chunk of bytes. Reading chunk length as int (4 bytes)... [OChannelBinaryServer]
2015-01-17 13:41:04:935 INFO /10.0.2.2:53356 - Read chunk lenght: 41 [OChannelBinaryServer]
2015-01-17 13:41:04:935 INFO /10.0.2.2:53356 - Reading 41 bytes... [OChannelBinaryServer]
2015-01-17 13:41:04:935 INFO /10.0.2.2:53356 - Read 41 bytes: TestEdge@Weight:1.3f,in:#11:-2,out:#11:-3 [OChannelBinaryServer]
2015-01-17 13:41:04:936 INFO /10.0.2.2:53356 - Reading byte (1 byte)... [OChannelBinaryServer]
2015-01-17 13:41:04:936 INFO /10.0.2.2:53356 - Read byte: 0 [OChannelBinaryServer]
2015-01-17 13:41:04:936 INFO /10.0.2.2:53356 - Reading chunk of bytes. Reading chunk length as int (4 bytes)... [OChannelBinaryServer]
2015-01-17 13:41:04:936 INFO /10.0.2.2:53356 - Read chunk lenght: 0 [OChannelBinaryServer]
2015-01-17 13:41:04:936 INFO /10.0.2.2:53356 - Reading 0 bytes... [OChannelBinaryServer]
2015-01-17 13:41:04:936 INFO /10.0.2.2:53356 - Read 0 bytes:  [OChannelBinaryServer]
2015-01-17 13:41:04:959 INFO /10.0.2.2:53356 - Writing byte (1 byte): 0 [OChannelBinaryServer]
2015-01-17 13:41:04:959 INFO /10.0.2.2:53356 - Writing int (4 bytes): 4 [OChannelBinaryServer]
2015-01-17 13:41:04:959 INFO /10.0.2.2:53356 - Writing int (4 bytes): 3 [OChannelBinaryServer]
2015-01-17 13:41:04:959 INFO /10.0.2.2:53356 - Writing short (2 bytes): 11 [OChannelBinaryServer]
2015-01-17 13:41:04:959 INFO /10.0.2.2:53356 - Writing cluster position (8 bytes) : -3.... [OChannelBinaryServer]
2015-01-17 13:41:04:960 INFO /10.0.2.2:53356 - Writing short (2 bytes): 11 [OChannelBinaryServer]
2015-01-17 13:41:04:960 INFO /10.0.2.2:53356 - Writing cluster position (8 bytes) : 1.... [OChannelBinaryServer]
2015-01-17 13:41:04:960 INFO /10.0.2.2:53356 - Writing short (2 bytes): 11 [OChannelBinaryServer]
2015-01-17 13:41:04:960 INFO /10.0.2.2:53356 - Writing cluster position (8 bytes) : -2.... [OChannelBinaryServer]
2015-01-17 13:41:04:960 INFO /10.0.2.2:53356 - Writing short (2 bytes): 11 [OChannelBinaryServer]
2015-01-17 13:41:04:960 INFO /10.0.2.2:53356 - Writing cluster position (8 bytes) : 0.... [OChannelBinaryServer]
2015-01-17 13:41:04:960 INFO /10.0.2.2:53356 - Writing short (2 bytes): 12 [OChannelBinaryServer]
2015-01-17 13:41:04:961 INFO /10.0.2.2:53356 - Writing cluster position (8 bytes) : -4.... [OChannelBinaryServer]
2015-01-17 13:41:04:961 INFO /10.0.2.2:53356 - Writing short (2 bytes): 12 [OChannelBinaryServer]
2015-01-17 13:41:04:961 INFO /10.0.2.2:53356 - Writing cluster position (8 bytes) : 0.... [OChannelBinaryServer]
2015-01-17 13:41:04:961 INFO /10.0.2.2:53356 - Writing int (4 bytes): 2 [OChannelBinaryServer]
2015-01-17 13:41:04:961 INFO /10.0.2.2:53356 - Writing short (2 bytes): 12 [OChannelBinaryServer]
2015-01-17 13:41:04:961 INFO /10.0.2.2:53356 - Writing cluster position (8 bytes) : 0.... [OChannelBinaryServer]
2015-01-17 13:41:04:962 INFO /10.0.2.2:53356 - Writing int (4 bytes): 1 [OChannelBinaryServer]
2015-01-17 13:41:04:962 INFO /10.0.2.2:53356 - Writing short (2 bytes): 11 [OChannelBinaryServer]
2015-01-17 13:41:04:962 INFO /10.0.2.2:53356 - Writing cluster position (8 bytes) : 0.... [OChannelBinaryServer]
2015-01-17 13:41:04:962 INFO /10.0.2.2:53356 - Writing int (4 bytes): 1 [OChannelBinaryServer]
2015-01-17 13:41:04:962 INFO /10.0.2.2:53356 - Flush [OChannelBinaryServer]
2015-01-17 13:41:04:963 INFO /10.0.2.2:53356 - Reading byte (1 byte)... [OChannelBinaryServer]
```

**Comment by lvca** [comment]
We already fix this in 2.0, where both storage behavior is identical. We're releasing 2.0-final on Tuesday, so I'd close this issue.

**Comment by GoorMoon** [comment]
@lvca 
We talk about this, and support for 1.7.* will be continue about one year from 2.0 release.
This is can be a problem for driver development.

**Comment by lvca** [comment]
We'll continue supporting 1.7 for clients with hot fixes. Back port this fix in 1.7 is way hard. I think the best is to put this in the driver limitations page.
